core/src/snapshot_packager_service.rs

```diff
@@ -233,107 +226,51 @@ mod tests {
-            .map(|i| Arc::new(AccountStorageEntry::new(&accounts_dir, 0, i, 10)))
-            .collect();
-
-        // Create some fake snapshot
-        let snapshots_paths: Vec<_> = (0..5)
-            .map(|i| {
-                let snapshot_file_name = format!("{i}");
-                let snapshots_dir = snapshots_dir.join(&snapshot_file_name);
-                fs::create_dir_all(&snapshots_dir).unwrap();
-                let fake_snapshot_path = snapshots_dir.join(&snapshot_file_name);
-                let mut fake_snapshot_file = OpenOptions::new()
-                    .read(true)
-                    .write(true)
-                    .create(true)
-                    .open(&fake_snapshot_path)
-                    .unwrap();
-
-                fake_snapshot_file.write_all(b"Hello, world!").unwrap();
-                fake_snapshot_path
-            })
-            .collect();
-
-        // Create directory of hard links for snapshots
-        let link_snapshots_dir = tempfile::tempdir_in(temp_dir).unwrap();
-        for snapshots_path in snapshots_paths {
-            let snapshot_file_name = snapshots_path.file_name().unwrap();
-            let link_snapshots_dir = link_snapshots_dir.path().join(snapshot_file_name);
-            fs::create_dir_all(&link_snapshots_dir).unwrap();
-            let link_path = link_snapshots_dir.join(snapshot_file_name);
-            fs::hard_link(&snapshots_path, link_path).unwrap();
-        }
+        let slot = 4;
 
-        // Create a packageable snapshot
-        let slot = 42;
-        let hash = SnapshotHash(Hash::default());
-        let archive_format = ArchiveFormat::TarBzip2;
-        let output_tar_path = snapshot_utils::build_full_snapshot_archive_path(
-            &full_snapshot_archives_dir,
+        let genesis_config = GenesisConfig::default();
+        let bank = snapshot_utils::create_snapshot_dirs_for_tests(
+            &genesis_config,
+            &bank_snapshots_dir,
+            slot,
             slot,
-            &hash,
-            archive_format,
         );
-        let snapshot_package = SnapshotPackage {
-            snapshot_archive_info: SnapshotArchiveInfo {
-                path: output_tar_path.clone(),
-                slot,
-                hash,
-                archive_format,
-            },
-            block_height: slot,
-            snapshot_links: link_snapshots_dir,
-            snapshot_storages: storage_entries,
-            snapshot_version: SnapshotVersion::default(),
-            snapshot_type: SnapshotType::FullSnapshot,
-            enqueued: Instant::now(),
-        };
-
-        // Make tarball from packageable snapshot
-        snapshot_utils::archive_snapshot_package(
-            &snapshot_package,
+
+        let bank_snapshot_info =
+            snapshot_utils::get_highest_bank_snapshot(&bank_snapshots_dir).unwrap();
+        let snapshot_storages = bank.get_snapshot_storages(None);
+        let archive_format = ArchiveFormat::TarBzip2;
```

Comment:
nit: I always try to declare variables as close to where they are used - better to move this after the `purge_old_bank_snapshots` call

Reply:
done.